test/boost/flat_mutation_reader_test.cc

```diff
@@ -727,6 +728,33 @@ SEASTAR_TEST_CASE(test_make_forwardable) {
     });
 }
 
+SEASTAR_TEST_CASE(test_make_nonforwardable) {
```

Comment:
We need test names that hint at what they are testing.

This test seems to stress the emitting logic for partition_end, so this should be in the name somehow.

Reply:
This test is not only about `partition_end`, it contains all the basic checks for `make_nonforwardable`, there were no tests for this function before. Each check is a separate region with its own comment. This way it's easy to reuse various helpers and test data. We can separate each check into its own test, but I don't see much value in it. The already existing `test_make_forwardable` is structured in the same way.

Reply:
We should add a test which subject this reader to the mutation source reader torture suite.

Reply:
That suite would have cough all these problems

Reply:
> We should add a test which subject this reader to the mutation source reader torture suite.

How to do it? Could you share an example?

Reply:
Search for `run_mutation_source_tests` in `test/boost/mutation_reader_test.cc` or `test/boost/flat_mutation_reader_test.cc` and you will see plenty of examples.
The gist is this:
* You create an underlying reader with the passed-in mutations. The most convenient is to create a memtable, dump the mutations into them, then create a memtable reader. You will always have to create this with `streamed_mutation::forwarding::yes`.
* Wrap this reader with the nonforwarding reader.
* When the test asks for forwarding reader, you can use `make_forwarding()` to make your nonforwarding reader forwarding one.

Reply:
This reader can't work with the general mutation_source tests, because it works only with a single partition. mutation source tests assume that the mutation source can accept multiple partitions.

Reply:
We could extract a battery of single-partition tests out of the general mutation source battery of tests.

Reply:
I added `test_make_nonforwardable_from_mutations_as_mutation_source`